jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/AbstractAdvanceRuntimeDataServiceImpl.java

```diff
@@ -0,0 +1,306 @@
+        for (Map.Entry<String, Object> var : variables.entrySet()) {
+            String alias = "V_TABLE_" + var.getKey();
+            String varName = var.getKey();
+            derivedTables.append("INNER JOIN (\n" +
+                                  "SELECT taskId, name, value \n" +
+                                  "FROM TaskVariableImpl \n" +
+                                  "WHERE type = 0\n" +
+                                 ") " + alias + " ON " + alias + ".taskId = task.id  \n");
+
+
+            addWhereExpression(where);
+            where.append(alias + ".NAME = :V_NAME_" + varName + " AND " + alias + ".VALUE= :V_VALUE_" + varName + "\n");
+        }
+
+        for (Map.Entry<String, Object> var : processVariables.entrySet()) {
+            String alias = "P_TABLE_" + var.getKey();
+            String varName = var.getKey();
+
+            derivedTables.append("LEFT JOIN (" +
+                                 "SELECT A1.processId, A1.processInstanceId, A1.variableId AS NAME, A1.value AS VALUE \n" +
+                                 "FROM VariableInstanceLog A1 \n" +
+                                 "LEFT JOIN VariableInstanceLog A2 ON A1.processId = A2.processId AND A1.processInstanceId = A2.processInstanceId AND A1.variableInstanceId = A2.variableInstanceId AND A2.log_date < A1.log_date  \n" +
+                                 "WHERE A2.log_date IS NULL \n" +
+                                 ") " + alias + " ON " + alias + ".processId = pil.processId AND " + alias + ".processInstanceId = pil.processInstanceId \n");
+
+            addWhereExpression(where);
+            where.append(alias + ".NAME = :P_NAME_" + varName + " AND " + alias + ".VALUE= :P_VALUE_" + varName + "\n");
+        }
+
+        if (!owners.isEmpty()) {
+            derivedTables.append("INNER JOIN ( \n" +
+                             "           SELECT DISTINCT po.task_id \n" +
+                             "           FROM PeopleAssignments_PotOwners po \n" +
+                             "           WHERE po.entity_id IN (:owners) \n" +
+                             "           GROUP BY po.task_id \n" +
+                             "           HAVING COUNT(po.entity_id) = :num_owners \n" +
+                                 ") AS pot ON pot.task_id = task.id ");
+        }
+
+        for (Map.Entry<String, Object> entry : attributes.entrySet()) {
+            addWhereExpression(where);
+            where.append(" task." + entry.getKey() + " = :ATTR_" + entry.getKey() + " ");
+        }
+
+        addWhereExpression(where);
+        where.append(" pil.processType = :processType ");
+
+        String procSQLString = "SELECT DISTINCT task.id FROM Task task INNER JOIN ProcessInstanceLog pil ON pil.processInstanceId = task.processInstanceId \n " + derivedTables + where + " ORDER BY id ASC ";
+        procSQLString += (queryContext.getCount() > 0) ? " LIMIT " + queryContext.getCount() + " OFFSET " + queryContext.getOffset() + "\n" : "";
```

Comment:
This native query is failing when using an Oracle database: **ORA-00933: SQL command not properly ended**

Seems it could be related to the pagination part of the SQL statement.